src/ib_py/cogs/channelorder.py

```diff
@@ -1,31 +1,209 @@
+import discord
 from discord.ext import commands
 
+from ..db.models import GuildSnapshot
+
 
 class ChannelOrder(commands.Cog):
     def __init__(self, bot: commands.Bot) -> None:
         self.bot = bot
 
     @commands.hybrid_group(aliases=["co"])
     async def channelorder(self, ctx: commands.Context):
+        """Commands for discord channel arrangement within categories."""
+        if ctx.invoked_subcommand is None:
+            await ctx.send_help(ctx.command)
+
+    def get_category(self, guild: discord.Guild, category_input: str | int) -> discord.CategoryChannel | None:
+        """
+        Retrieves a category from a guild by name or ID.
         """
-        Commands for discord channel arrangement within categories.
+        # Try ID
+        try:
+            category_id = int(category_input)
+            category = discord.utils.get(guild.categories, id=category_id)
+            if category:
+                return category
+        except ValueError:
+            pass  # Not an ID, so treat as name
+
+        # Try by name (case-insensitive)
+        category = discord.utils.find(lambda c: c.name.lower() == str(category_input).lower(), guild.categories)
+        return category
+
+    def draw_embed(self, title: str, fields: list[tuple[str, str, bool]], color=discord.Color.blue()):
         """
-        raise NotImplementedError("Command requires implementation and permission set-up.")
+        Helper function to draw an embed with the given title, fields, and color.
+        
+        Args:
+            title (str): The title of the embed.
+            fields (list[tuple[str, str, bool]]): A list of fields where each field is a tuple (name, value, inline).
+            color (discord.Color, optional): The color of the embed. Defaults to discord.Color.blue().
 
-    @channelorder.command()
-    async def snapshot(self, ctx: commands.Context):
+        Returns:
+            discord.Embed: The constructed embed object.
         """
-        Save the arrangement of channels in a category.
+        embed = discord.Embed(title=title, color=color)
+        for name, value, inline in fields:
+            embed.add_field(name=name, value=value, inline=inline)
+        return embed
+
+    def create_snapshot_embed(self, category, channels):
         """
-        raise NotImplementedError("Command requires implementation and permission set-up.")
+        Helper function to create a snapshot embed for a given category.
+        Channels are listed on separate lines and mentioned using <#id>.
+        """
+        fields = []
+        if channels:
+            # Mention each channel on a new line
+            channel_mentions = "\n".join(f"<#{ch.id}>" for ch in channels)
+        else:
+            channel_mentions = "None"
+
+        fields.append(("Channels", channel_mentions, False))
+        return self.draw_embed(title=f"📸 Snapshot for {category.name}", fields=fields)
```

Comment:
Is it necessary to separate the drawing of the embed to a separate method? This method can be greatly condensed while improving readability:
```py
title = f"📸 Snapshot for {category.name}"
channels = "\n".join(ch.mention for ch in channels) or "Empty category."
return discord.Embed(title=title, description=channels, color=discord.Color.blue())
```